### DISCUSSION TITLE
WPF - Chromecast Support

### DISCUSSION THREAD

**Discussion by wm-v-michaels** [cefsharp/CefSharp#4343] [General]
I've seen a few threads on how to get this working, but anecdotally support has either stopped, or was never there to begin with.   I've read that using some combination of the below works/has worked:

 settings.CefCommandLineArgs.Add("disable-gpu");

                settings.UserAgent = "CefSharp Browser" + Cef.CefSharpVersion; // Example User Agent
                settings.CefCommandLineArgs.Add("renderer-process-limit", "4");
                //settings.CefCommandLineArgs.Add("renderer-startup-dialog", "renderer-startup-dialog");
                //settings.CefCommandLineArgs.Add("disable-gpu", "1");

                // trying to make chromecast work
                 settings.CefCommandLineArgs.Add("enable-media-stream", "1"); //Enable WebRTC
                settings.CefCommandLineArgs.Add("enable-npapi", "1");
                settings.CefCommandLineArgs.Add("no-sandbox", "1");
                settings.CefCommandLineArgs.Add("use-fake-ui-for-media-stream", "1");
                settings.CefCommandLineArgs.Add("enable-usermedia-screen-capturing", "1");

                settings.CefCommandLineArgs.Add("disable-gpu-vsync", "1");
                settings.CefCommandLineArgs.Add("disable-gpu-compositing", "1");
                settings.CefCommandLineArgs.Add("disable-software-rasterizer", "1");
                
                settings.CefCommandLineArgs.Add("no-proxy-server", "1");  //Don't use a proxy server, always make direct connections. Overrides any other proxy server flags that are passed.
                settings.CefCommandLineArgs.Add("debug-plugin-loading", "1");  //Dumps extra logging about plugin loading to log file.
                settings.CefCommandLineArgs.Add("disable-plugins-discovery", "1"); //Disable discovering third-party plugins. Effectively loading only ones shipped with the browser plus third-party ones as specified by --extra-plugin-dir and --load-plugin switches
                settings.CefCommandLineArgs.Add("off-screen-rendering-enabled", "1");
                //Disables the DirectWrite font rendering system on windows.
                //Possibly useful when experiencing blury fonts.
                settings.CefCommandLineArgs.Add("disable-direct-write", "1");

At current,  I'm able to see the "Play On TV" button when using Chrome, but not when embedding CEFSharp (with touch support) in a WPF App.

Is this support planned?

**Comment by amaitland** [comment]
> Is this support planned?

Technically I believe `CEF` already has support for [Media Routing](https://bitbucket.org/chromiumembedded/cef/issues/2900/add-mediarouter-support-for-cast-receiver). 

The `CefMediaRouter` implementation isn't currently exposed through `CefSharp`, so you'd possible need to expose that and related classes/structs.

Not something I've ever tried. Best place to ask follow up questions would be on [ceforum](https://www.magpcss.org/ceforum/index.php) (make sure to post a link back here for others to follow).

I'd suggest testing with the [CEF Sample Application](https://github.com/cefsharp/CefSharp/blob/master/CONTRIBUTING.md#cefsharp-vs-chromium-embedded-frameworkcef) initially, get it working there then you can work on getting it working in `CefSharp`.

Best of luck!

**Reply by wm-v-michaels** [reply]
Thanks for the breadcrumbs.  I'll take a look at bringing CefMediaRouter to the surface and all of its friends :) If I manage to make it work, I'll try to create a pull request and share the love.

Given there hasn't been much/any interest here, do you think it's worth other people's time, or should I just fork it?

**Reply by amaitland** [reply]
> To test: Navigate to http://tests/media_router in cefclient and follow the on-screen instructions.

Confirm that CEF supports the features you are after first. Ask on ceforum if you need clarification.

You are welcome to submit a `PR` exposing any additional `CEF` classes.

> Given there hasn't been much/any interest here, do you think it's worth other people's time, or should I just fork it?

Maintaining your own fork is a big undertaking if you want to keep up with `Chromium` releases.
